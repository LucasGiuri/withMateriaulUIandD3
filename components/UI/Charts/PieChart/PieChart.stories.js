import { storiesOf } from '@storybook/react';
import PieChart from './PieChart';

const data = [
  {name: 'Shalion 1', value: 1},
  {name: 'Shalion 2', value: 2},
  {name: 'Shalion 3', value: 3},
  {name: 'Shalion 4', value: 4},
  {name: 'Shalion 5', value: 4},
];

storiesOf('PieChart')
  .add('PieChart default', () => (
    <PieChart data={data}/>
  ))
