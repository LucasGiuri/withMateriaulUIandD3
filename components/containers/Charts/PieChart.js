import PieChartComponent from '../../UI/Charts/PieChart/PieChart';

const PieChart = ({data}) => {
  let newData = [];
  data.map(product => {
    newData.push({name: product.name, value: product.count})
  });

  return (
    <PieChartComponent data={newData} />
  )
}

export default PieChart;