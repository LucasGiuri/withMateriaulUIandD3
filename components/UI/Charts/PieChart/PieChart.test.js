import React from 'react';
import renderer from 'react-test-renderer';
import PieChart from './PieChart';
import 'jest-styled-components';

describe('PieChart', () => {
  const data = [
    {name: 'Shalion 1', value: 1},
    {name: 'Shalion 2', value: 2},
    {name: 'Shalion 3', value: 3},
    {name: 'Shalion 4', value: 4},
    {name: 'Shalion 5', value: 4},
  ];

  it('PieChart basic render snapshoot', () => {
    const component = renderer.create(<PieChart  data={data} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
