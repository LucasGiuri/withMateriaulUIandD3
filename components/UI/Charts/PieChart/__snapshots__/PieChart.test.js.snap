// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PieChart PieChart basic render snapshoot 1`] = `
<div
  className="recharts-wrapper"
  style={
    Object {
      "cursor": "default",
      "height": 400,
      "position": "relative",
      "width": 400,
    }
  }
>
  <svg
    className="recharts-surface"
    cx="50%"
    cy="50%"
    height={400}
    version="1.1"
    viewBox="0 0 400 400"
    width={400}
  >
    <defs>
      <clipPath
        id="recharts1-clip"
      >
        <rect
          height={390}
          width={390}
          x={5}
          y={5}
        />
      </clipPath>
    </defs>
    <g
      className="recharts-layer recharts-pie"
    >
      <g
        className="recharts-layer"
      >
        <g
          className="recharts-layer recharts-pie-sector"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <g
          className="recharts-layer recharts-pie-sector"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <g
          className="recharts-layer recharts-pie-sector"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <g
          className="recharts-layer recharts-pie-sector"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
        <g
          className="recharts-layer recharts-pie-sector"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
      </g>
    </g>
  </svg>
</div>
`;
